sections/appendix/problems.tex

```diff
@@ -0,0 +1,66 @@
+We then changed our algorithm to only consider documents that shared topics to the original documents, decreasing our time to $D*T_d*D_t$, where both of the new variables are reduced compared to the originals. This reduced the calculation time to be estimated as ~48 days.
+We also made the calculation multithreaded in order to reduce computation time. Estimated calculation time is reduced to ~2,4 days using 8 threads.
+Lastly, we changed to construct our matrix using the \emph{lil\_matrix} format, which has faster insertions. This last change made the construction much more manageable.
+
+
+\todo[inline]{Nedenstående problem skal uddybes, hvis det beholdes. Skal også have sit eget afsnit separat fra Adjacency Matrix Construction}
+The function get\_document\_topics might be needing the whole corpus since whenever we give it the query it returns the same two topics. 
+These two topics are probably the two most general topics within the model which might spell trouble since we don't know how to remove these from the model.
+
+\subsubsection{Dirichlet distributions only produce non-zero values}
+We use document-topic and topic-word distributions produced by the Dirichlet distributions to construct our adjacency matrix and to convert queries into topic distributions. 
+However as part of how \gls{lda} is designed it has no zeros in its $\theta$ and $\beta$ values, only extremely low numbers.
+This is not optimal for constructing an adjacency matrix, as it would result in a fully connected graph.
+Adjusting $\alpha$ and $\eta$ will make the low values lower, and the high values higher, but this does not make the problem disappear.
+We have therefore introduced thresholds.
+These thresholds reduce all values lower than the threshold to zero in the document-topic matrix and topic-word matrix.
+The new problem introduced by introducing thresholds is that the exact values of the thresholds are now added as extra hyper-parameters for our solution.
+
+\subsubsection{Unable to reverse engineer the \gls{lda} generative process}
+In the generative process of \gls{lda}, a document $d$ is assumed to have a certain underlying topic-distribution $\theta_d$ and a topic $k$ is assumed to have a certain underlying word-distribution $\beta_k$.
+In \gls{lda}, words are generated for documents by choosing a topic from the topic-distribution of the document and then choosing a word based on the word-distribution of that topic.
+
+Our initial idea for query handling was to reverse engineer this process.
+For each word in a query, we would convert a word into a topic-distribution and then convert that topic-distribution into a document distribution.
+This would be done using the same tables that were used to create the words $\theta$ and $\varphi$, but by accessing columns rather than rows, which also meant that the array of numbers we received had to be converted into distributions, as only the rows of these matrices were guaranteed to sum to 1.
```

Comment:
jeg tror jeg har en todo med at omskrive vores initial idé baseret på det som stod i query handling

Reply:
```suggestion
This would be done using the same tables that were used to create the words $\theta$ and $\varphi$, but by accessing columns rather than rows, which also meant that the array of numbers we received had to be converted into distributions, as only the rows of these matrices were guaranteed to sum to 1.
\todo{'the words $\theta$ and $\varphi$' sounds bad and confusing}
```